### PR TITLE
ScalafmtDynamic: throw once if can't resolve config

### PR DESCRIPTION
**Pull request by kitbellew**
Intellij invokes the createSession method directly and doesn't expect the null value we return there, so throw ConfigError instead.

Similarly, in the Runner, change the implementation of MyInstanceSession to verify the config first, so that the config parsing exception is not thrown for every file we are trying to format.

### CODE REVIEW

File: scalafmt-cli/src/main/scala/org/scalafmt/cli/ScalafmtDynamicRunner.scala [hunk 1]
```diff
@@ -27,12 +31,15 @@ object ScalafmtDynamicRunner extends ScalafmtRunner {
       .withReporter(reporter)
       .withRespectProjectFilters(false)
 
-    val session = scalafmtInstance match {
+    val session = Try(scalafmtInstance match {
```
Comment (poslegm) [inline scalafmt-cli/src/main/scala/org/scalafmt/cli/ScalafmtDynamicRunner.scala:34]: Why not just try/catch here?

File: scalafmt-cli/src/main/scala/org/scalafmt/cli/ScalafmtDynamicRunner.scala [hunk 2]
```diff
@@ -79,6 +86,8 @@ object ScalafmtDynamicRunner extends ScalafmtRunner {
 
   private final class MyInstanceSession(opts: CliOptions, instance: Scalafmt)
       extends ScalafmtSession {
+    // check config first
+    locally { instance.format(opts.configPath, opts.configPath, "") }
```
Comment (poslegm) [inline scalafmt-cli/src/main/scala/org/scalafmt/cli/ScalafmtDynamicRunner.scala:90]: I can't understand how `locally` helps here

Comment (kitbellew) [reply scalafmt-cli/src/main/scala/org/scalafmt/cli/ScalafmtDynamicRunner.scala:90]: in that case, i will remove. perhaps i am still getting used to being able to invoke methods in scala which are not part of another method.

File: scalafmt-cli/src/main/scala/org/scalafmt/cli/ScalafmtDynamicRunner.scala (hunk 1 above)
Comment (kitbellew) [reply scalafmt-cli/src/main/scala/org/scalafmt/cli/ScalafmtDynamicRunner.scala:34]: hmm... good idea! :)

**Pull request merged**